### PR TITLE
[Student][MBL-13094] Analytics for 6.6

### PR DESCRIPTION
**Pull request by Blackchai**
I haven't had a chance to verify all of the events yet on firebase debug, but it should be ready for code review. Ping me if you want to see the spreadsheet with the names for the analytics.

Atm, I'm not stoked on bundling little verify tests in with the existing unit tests, but it seemed like the best way to keep coverage up, test the analytic calls that are in effect handlers, while simultaneously not adding duplicate tests that just test analytics.

I'm also not stoked on how I added analytics around the tabs in the student speedgrader page, open to suggestions!

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1005501&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/AssignmentDetailsEffectHandler.kt [hunk 1]
```diff
@@ -21,13 +21,12 @@ import android.content.Context
...
 import com.instructure.canvasapi2.managers.SubmissionManager
-import com.instructure.canvasapi2.models.Assignment
-import com.instructure.canvasapi2.models.DiscussionTopic
-import com.instructure.canvasapi2.models.LTITool
+import com.instructure.canvasapi2.models.*
 import com.instructure.canvasapi2.utils.*
 import com.instructure.canvasapi2.utils.weave.StatusCallbackError
 import com.instructure.canvasapi2.utils.weave.awaitApiResponse
 import com.instructure.student.Submission
+import com.instructure.student.analytics.*
```
Comment (JordanMarshall) [inline apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/AssignmentDetailsEffectHandler.kt]: It looks like this package doesn't exist and is causing the build failure for Student

File: apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/submissionDetails/content/emptySubmission/ui/SubmissionDetailsEmptyContentView.kt [hunk 2]
```diff
@@ -209,4 +219,8 @@ class SubmissionDetailsEmptyContentView(
         // Not run on main thread of fragment host by default, so force it to run on UI thread
         (context as Activity).runOnUiThread { (context as Activity).onBackPressed() }
     }
+
+    private fun logAnalytics(event: String) {
```
Comment (JordanMarshall) [inline apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/submissionDetails/content/emptySubmission/ui/SubmissionDetailsEmptyContentView.kt]: Similar to `EffectHandler`, would it make sense to add a `logAnalytics` function to the base `MobiusView` class?

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt [hunk 3]
```diff
@@ -34,16 +34,64 @@ object Analytics {
...
+    const val SUBMIT_TEXTENTRY_SELECTED = "submit_textentry_selected"
+    const val SUBMIT_MEDIARECORDING_SELECTED = "submit_mediarecording_selected"
+    const val SUBMIT_STUDIO_SELECTED = "submit_studio_selected"
+    const val SUBMISSION_CELL_SELECTED = "submission_cell_selected"
+    const val SUBMISSION_COMMENTS_SELECTED = "assignment_submit_selected"
+    const val SUBMISSION_COMMENTS_TEXT_REPLY = "submission_comments_text_reply"
+    const val SUBMISSION_COMMENTS_MEDIA_REPLY = "submission_comments_media_reply"
+    const val SUBMISSION_RUBRIC_SELECTED = "submission_rubric_selected"
+    const val SUBMISSION_FILES_SELECTED = "submission_files_selected"
+    const val SUBMISSION_ANNOTATION_SELECTED = "SUBMISSION_RUBRIC_SELECTED"
```
Comment (JordanMarshall) [inline libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt]: Is this right?

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt [hunk 4]
```diff
@@ -34,16 +34,64 @@ object Analytics {
...
+    const val ASSIGNMENT_DETAIL_DISCUSSIONLAUNCH = "assignment_detail_discussionlaunch"
+    const val ASSIGNMENT_SUBMIT_SELECTED = "assignment_submit_selected"
+    const val ASSIGNMENT_LAUNCHLTI_SELECTED = "assignment_launchlti_selected"
+    const val SUBMIT_FILEUPLOAD_SELECTED = "submit_fileupload_selected"
+    const val SUBMIT_ONLINEURL_SELECTED = "submit_onlineurl_selected"
+    const val SUBMIT_TEXTENTRY_SELECTED = "submit_textentry_selected"
+    const val SUBMIT_MEDIARECORDING_SELECTED = "submit_mediarecording_selected"
+    const val SUBMIT_STUDIO_SELECTED = "submit_studio_selected"
+    const val SUBMISSION_CELL_SELECTED = "submission_cell_selected"
+    const val SUBMISSION_COMMENTS_SELECTED = "assignment_submit_selected"
```
Comment (JordanMarshall) [inline libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt]: Is this right?

Comment (Blackchai) [reply libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt]: Wow lots of fails here, my bad. This is what I get for switching on and off this branch like 5 times 🤦‍♂ 

Thanks for the good feedback on this PR.

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt (hunk 3 above)
Comment (Blackchai) [reply libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/Analytics.kt]: 🤦‍♂

File: apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/submissionDetails/content/emptySubmission/ui/SubmissionDetailsEmptyContentView.kt (hunk 2 above)
Comment (Blackchai) [reply apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/submissionDetails/content/emptySubmission/ui/SubmissionDetailsEmptyContentView.kt]: 👍

File: apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/AssignmentDetailsEffectHandler.kt [hunk 5]
```diff
@@ -180,6 +179,8 @@ class AssignmentDetailsEffectHandler(val context: Context, val assignmentId: Lon
                 }
             } else null
 
+            Analytics.logEvent(getAnalyticsString(quizResult, assignmentResult))
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/AssignmentDetailsEffectHandler.kt]: 👍

File: apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/AssignmentDetailsEffectHandler.kt (hunk 1 above)
Comment (Blackchai) [reply apps/student/src/main/java/com/instructure/student/mobius/assignmentDetails/AssignmentDetailsEffectHandler.kt]: 🤦‍♂

**Pull request merged**